### ISSUE TITLE
gnomeTheme.hideSingleTab doesn't work with simple tab groups extension

### ISSUE THREAD

**Issue by RTSAjwad** [rafaelmardojai/firefox-gnome-theme#709]
**Describe the bug**
When enabling gnomeTheme.hideSingleTab, it still shows the single tab if you are using the simple tab groups extension. This is because there are technically multiple tabs but they are hidden. May this has to do with hidden tabs in general.

**To Reproduce**

1. Use simple tab groups extension
2. Enable gnomeTheme.hideSingleTab
4. Make group with only a single tab in it the active group whilst there are other inactive groups with tabs in them
5. Tab bar is still showing with one tab

**Expected behavior**
The tab bar should be hidden with one tab even if there are other tab groups with tabs in them

**Screenshots**
![Screenshot from 2023-11-25 20-47-11](https://github.com/rafaelmardojai/firefox-gnome-theme/assets/55037109/ac1f9107-2fa9-4d60-954d-0573c8087d8a)
Here you can see with this screenshot that the single tab is showing even though gnomeTheme.hideSingleTab is turned on
![Screenshot from 2023-11-25 20-48-39](https://github.com/rafaelmardojai/firefox-gnome-theme/assets/55037109/68b33936-9cff-45ad-b5ed-d664f67c1898)
In this screenshot you can see that there are technically other tabs but they are in another group

**Setup information**
 - OS: Fedora
 - Installation method: Flatpak
 - Firefox version: 120.0
 - Theme version: master
 - Optional features enabled: e.g. gnomeTheme.hideSingleTab

**Comment by rafaelmardojai** [comment]
Yes, this is a known limitation and not really fixable, I should probably document this in the README.

The problem is that when extensions hide tabs, they're still there in the tab bar, just visually hidden, so there's no way to know from CSS that there's only one tab that's not hidden.

Before Firefox 110, we had the `first-visible-tab` and `last-visible-tab` attributes in tabs, and if a tab had both true it could be hidden, but Mozilla removed them.

**Comment by RTSAjwad** [comment]
Ahh I understand now. Thanks for the information. Also you've done a great job with this project because it work fantastically!

**Comment by SamyzKhalil** [comment]
Since `:has` is now [supported](https://www.mozilla.org/en-US/firefox/121.0/releasenotes/#note-789915) in Firefox 121, it is possible.

In `firefox-gnome-theme/theme/parts/tabsbar.css`

replace 
```css 
#tabbrowser-tabs tab:only-of-type
```
with 
```css
#tabbrowser-tabs:not(:has(tab:not([hidden="true"]) + tab:not([hidden="true"]))) tab
```
Not sure if the selector could be simplified but it works.
